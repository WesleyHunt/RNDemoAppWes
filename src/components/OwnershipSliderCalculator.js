import React, { useState, useEffect } from 'react';
import { View, Text, StyleSheet, Slider, TouchableOpacity } from 'react-native';
import currency from 'currency.js';
import {Feather} from '@expo/vector-icons';

export const CalculateOwnership = (initialInvestment, monthlyInvestment, yearsRenting) => {
    return (parseInt(initialInvestment, 10) + parseInt(monthlyInvestment, 10) * parseInt(yearsRenting, 10) * 12) ?? "error"
}

export const RoundToNearestMultiplier = (value, multiplier) => {
    return (Math.round(value / multiplier) * multiplier)
}

export const FormatToCurrency = (value) =>{
    return currency(value, {precision: '0'}).format()
}
const multiplier_25 = 25
const multiplier_5 = 5

const OwnershipSliderCalculator = ({ rent, initialMin, initialMax, monthlyMax, yearsMax }) => {
    const [initialInvestment, setInitialInvestment] = useState(0)
    const [initialSlider, setInitialSlider] = useState(0)
    const [monthlyInvestment, setMonthlyInvestment] = useState(0)
    const [monthlySlider, setMonthlySlider] = useState(0)
    const [yearsWithCompany, setYearsWithCompany] = useState(1)
    const [ownershipValue, setOwnershipValue] = useState(0)
    useEffect(() => {
        setOwnershipValue(CalculateOwnership(initialInvestment, monthlyInvestment, yearsWithCompany))
    });

    
    return (
        <View>
            <Text style={[styles.greyBoldText, { marginTop: 15, marginLeft: 20 }]}>MONTHLY PAYMENT</Text>
            <View style={{ flexDirection: 'row', marginLeft: 20, marginTop: 7 }}>
                <Text style={styles.blackText}>${FormatToCurrency(rent)} rent +</Text>
                <Text style={[styles.blueText, { marginLeft: 5 }]}>${FormatToCurrency(initialInvestment)} investment</Text>
            </View>
            <View style={{ backgroundColor: "#d3d3d3", height: 1, marginTop: 3 }} />
            <View style={{ backgroundColor: "#d3d3d3", height: 1, marginTop: 2 }} />
            <View style={{ marginTop: 35, marginLeft: 30, flexDirection: 'row' }}>
                <Text style={[styles.greyBoldText, styles.strikeThrough]}>SECURITY DEPOSIT</Text>
                <Text style={styles.greyBoldText}> &#x2192; INITIAL INVESTMENT</Text>
            </View>
            <View style={styles.buttonContainer}>
                <TouchableOpacity onPress={() =>{ 
                ((initialInvestment-multiplier_25) < initialMin) ? setInitialInvestment(initialMin) : setInitialInvestment(initialInvestment-multiplier_25)
                setInitialSlider(initialInvestment)}
                }>
                <Feather name="minus" style={styles.iconStyle}/>
                </TouchableOpacity>
                <Text style={styles.blackText}>${FormatToCurrency(initialInvestment)}</Text>
                <TouchableOpacity onPress={() => { 
                ((initialInvestment+multiplier_25) > initialMax) ? setInitialInvestment(initialMax) : setInitialInvestment(initialInvestment+multiplier_25)
                setInitialSlider(initialInvestment)}
                }>
                <Feather name="plus" style={styles.iconStyle}/>
                </TouchableOpacity>
            </View>
            
            <Slider
                style={{ alignSelf: 'stretch', marginLeft: 25, marginRight: 25, height: 40 }}
                value={initialSlider}
                minimumValue={initialMin ?? 0}
                maximumValue={initialMax ?? 12000}
                minimumTrackTintColor="#00CFCC"
                maximumTrackTintColor="#919191"
                thumbTintColor="#00008b"
                onValueChange={value => {
                    setInitialInvestment(RoundToNearestMultiplier(value, multiplier_25))
                }
                }
            />

            <Text style={[styles.greyBoldText, { marginLeft: 30, marginTop: 15 }]}>MONTHLY INVESTMENT</Text>
            <View style={styles.buttonContainer}>
                <TouchableOpacity onPress={() =>{ 
                ((monthlyInvestment-multiplier_5) < 0) ? setMonthlyInvestment(0) : setMonthlyInvestment(monthlyInvestment-multiplier_5)
                setMonthlySlider(monthlyInvestment)}
                }>
                <Feather name="minus" style={styles.iconStyle}/>
                </TouchableOpacity>
                <Text style={styles.blackText}>${monthlyInvestment}</Text>
                <TouchableOpacity onPress={() => { 
                ((monthlyInvestment+multiplier_5) > monthlyMax) ? setMonthlyInvestment(monthlyMax) : setMonthlyInvestment(monthlyInvestment+multiplier_5)
                setMonthlySlider(monthlyInvestment)}
                }>
                <Feather name="plus" style={styles.iconStyle}/>
                </TouchableOpacity>
            </View>

            <Slider
                style={{ alignSelf: 'stretch', marginLeft: 25, marginRight: 25, height: 40 }}
                value={monthlySlider}
                minimumValue={0}
                maximumValue={monthlyMax ?? 7000}
                minimumTrackTintColor="#00CFCC"
                maximumTrackTintColor="#919191"
                thumbTintColor="#00008b"
                onValueChange={value => {
                    setMonthlyInvestment(RoundToNearestMultiplier(value, multiplier_5))
                }
                }
            />

            <Text style={[styles.greyBoldText, { marginLeft: 30, marginTop: 15 }]}>YEARS WITH UP&UP</Text>
            <Text style={[styles.blackText, { alignSelf: 'center', paddingTop: 10 }]}>{yearsWithCompany} years</Text>
            <Slider
                style={{ alignSelf: 'stretch', marginLeft: 25, marginRight: 25, height: 40 }}
                value={yearsWithCompany}
                minimumValue={1}
                maximumValue={yearsMax ?? 5}
                minimumTrackTintColor="#00CFCC"
                maximumTrackTintColor="#919191"
                thumbTintColor="#00008b"
                onSlidingComplete={value => {
                    var whole = Math.round(value)
                    setYearsWithCompany(whole)
                }
                }
            />
            <Text style={[styles.greyBoldText, { marginLeft: 30, marginTop: 15 }]}>OWNERSHIP VALUE</Text>
            <Text style={[styles.blueText, { marginTop: 15, fontSize: 28, alignSelf: 'center' }]}>${FormatToCurrency(ownershipValue)}</Text>
            <Text style={[styles.greyText, { marginLeft: 30, marginRight: 30, textAlign: 'center' }]}>
                Use your ${FormatToCurrency(ownershipValue)} ownership stake as a down payment on any home, transfer it to your next up&up rental, or cash out.</Text>

        </View>
    )
}


const styles = StyleSheet.create({
    greyBoldText: {
        fontWeight: 'bold',
        fontSize: 15,
        color: '#949494'
    },
    greyText: {
        fontSize: 12,
        color: '#949494'
    },
    strikeThrough: {
        textDecorationLine: 'line-through',
        textDecorationStyle: 'solid'
    },
    blackText: {
        fontSize: 21,
        color: '#000000'
    },
    blueText: {
        fontSize: 21,
        color: '#00CFCC'
    },
    iconStyle: {
        fontSize: 35,
        alignSelf: 'center',
        marginHorizontal: 10
    },
    buttonContainer: {
        flexDirection: 'row', 
        justifyContent:'space-between', 
        alignItems: 'flex-end', 
        marginTop: 15,
        marginLeft: 40,
        marginRight: 40
    }
})

export default OwnershipSliderCalculator