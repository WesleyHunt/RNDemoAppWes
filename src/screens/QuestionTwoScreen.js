import React, { useState } from 'react'
import { View, Text, StyleSheet, ScrollView } from 'react-native'
import { TouchableOpacity } from 'react-native-gesture-handler'
import  OwnershipSliderCalculator from '../components/OwnershipSliderCalculator'

const QuestionTwoScreen = ({initialMin, initialMax, monthlyMax, yearsMax}) => {
    const [initialInvestment, setInitialInvestment] = useState('0')
    const [monthlyInvestment, setMonthlyInvestment] = useState('0')
    const [yearsWithCompany, setYearsWithCompany] = useState('0')
    const [ownershipValue, setOwnershipValue] = useState('0')

    return (
        <View style={{flex: 1}}>
            <Text style={{marginBottom: 10, marginLeft: 10}}>
            Component called with:: rent: 2000, Initial min investment: 1200, initial max: 12000, 
            monthly max investment: 600, years max: 5
            </Text>
            <View style={{ backgroundColor: "#d3d3d3", height: 7, marginTop: 1 }} />
            <ScrollView>
                 <OwnershipSliderCalculator rent={2000} initialMin={1200} initialMax={12000} monthlyMax={600} yearsMax={5}/>
            </ScrollView>
        </View>
    )
}


const styles = StyleSheet.create({
    greyBoldText: {
        fontWeight: 'bold',
        fontSize: 15,
        color: '#949494'
    },
    greyText: {
        fontSize: 12,
        color: '#949494'
    },
    strikeThrough: {
        textDecorationLine: 'line-through',
        textDecorationStyle: 'solid'
    },
    blackText: {
        fontSize: 21,
        color: '#000000'
    },
    blueText: {
        fontSize: 21,
        color: '#00CFCC'
    }
})

export default QuestionTwoScreen