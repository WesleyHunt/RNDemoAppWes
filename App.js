import { createAppContainer } from 'react-navigation';
import { createStackNavigator } from 'react-navigation-stack';
import HomeScreen from './src/screens/HomeScreen'
import QuestionOneScreen from './src/screens/QuestionOneScreen'
import QuestionTwoScreen from './src/screens/QuestionTwoScreen'

const navigator = createStackNavigator({
    HomeScreen: HomeScreen,
    QuestionOneScreen: QuestionOneScreen,
    QuestionTwoScreen: QuestionTwoScreen
  },
  {
    initialRouteName: 'HomeScreen',
    defaultNavigationOptions: {
      title: 'Wes App'
    }
  }
)

export default createAppContainer(navigator)