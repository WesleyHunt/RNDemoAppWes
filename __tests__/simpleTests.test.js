import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';
import QuestionTwoScreen from '../src/screens/QuestionTwoScreen';
import OwnershipSliderCalculator, {CalculateOwnership, RoundToNearestMultiplier} from '../src/components/OwnershipSliderCalculator';

//child check test
describe('<App />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree.children.length).toBe(1)
  })

  //Check base page snapshot has not changed
  it('renders page correctly', () => {
    const tree = renderer.create(<QuestionTwoScreen />).toJSON();
    expect(tree).toMatchSnapshot()
  })

  //check component has not changed
  it('renders calculator component correctly', () => {
    const tree = renderer.create(<OwnershipSliderCalculator />).toJSON();
    expect(tree).toMatchSnapshot()
  })

   //check component has not changed
   it('renders calculator component with props correctly', () => {
    const tree = renderer.create(<OwnershipSliderCalculator initialInvestment={1200} monthlyInvestment={100} yearsWithCompany={5}/>).toJSON();
    expect(tree).toMatchSnapshot()
  })

})

describe('Calculation functions', () => {
  it('returns a non null / invalid value', () => {
    expect(CalculateOwnership(null, null, null)).toStrictEqual(expect.anything())
    expect(CalculateOwnership(null, 100, null)).toStrictEqual(expect.anything())
    expect(CalculateOwnership(null, "helloworld", null)).toStrictEqual(expect.anything())
    expect(CalculateOwnership(1,2,3)).toStrictEqual(expect.anything())
    expect(CalculateOwnership('1', '2', '3')).toStrictEqual(expect.anything())
    expect(CalculateOwnership("hellow", "helloworld","world")).toStrictEqual(expect.anything())
  })

  it('returns expected calculated value', () => {
    /**redundant check: function calculateOwnership (initialInvestment, monthlyInvestment, monthsRenting) => {
    return initialInvestment + (monthlyInvestment * monthsRenting)
    return 1500 + (100 * yearsRenting*12)
    return 1500 + (100 * 1*12) = 2700
    **/
   expect(CalculateOwnership(1500,100,1)).toStrictEqual(2700)
  })

  it('round function returns expected value', () => {
    expect(RoundToNearestMultiplier(101, 25)).toStrictEqual(100)
    expect(RoundToNearestMultiplier(100, 25)).toStrictEqual(100)
    expect(RoundToNearestMultiplier(99, 25)).toStrictEqual(100)
    expect(RoundToNearestMultiplier(112, 25)).toStrictEqual(100)
    expect(RoundToNearestMultiplier(113, 25)).toStrictEqual(125)
  })

})