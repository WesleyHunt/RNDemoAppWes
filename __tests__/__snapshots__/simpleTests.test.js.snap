// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders calculator component correctly 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#949494",
          "fontSize": 15,
          "fontWeight": "bold",
        },
        Object {
          "marginLeft": 20,
          "marginTop": 15,
        },
      ]
    }
  >
    MONTHLY PAYMENT
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 20,
        "marginTop": 7,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 21,
        }
      }
    >
      $
       rent +
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#00CFCC",
            "fontSize": 21,
          },
          Object {
            "marginLeft": 5,
          },
        ]
      }
    >
      $
      0
       investment
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#d3d3d3",
        "height": 1,
        "marginTop": 3,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#d3d3d3",
        "height": 1,
        "marginTop": 2,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 30,
        "marginTop": 35,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#949494",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "textDecorationLine": "line-through",
            "textDecorationStyle": "solid",
          },
        ]
      }
    >
      SECURITY DEPOSIT
    </Text>
    <Text
      style={
        Object {
          "color": "#949494",
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
       → INITIAL INVESTMENT
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 21,
        },
        Object {
          "alignSelf": "center",
          "paddingTop": 10,
        },
      ]
    }
  >
    $
    0
  </Text>
  <RCTSlider
    disabled={false}
    enabled={true}
    maximumTrackTintColor="#919191"
    maximumValue={12000}
    minimumTrackTintColor="#00CFCC"
    minimumValue={0}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onSlidingComplete={null}
    onStartShouldSetResponder={[Function]}
    onValueChange={[Function]}
    step={0}
    style={
      Array [
        Object {
          "height": 40,
        },
        Object {
          "alignSelf": "stretch",
          "height": 40,
          "marginLeft": 25,
          "marginRight": 25,
        },
      ]
    }
    thumbTintColor="#00008b"
    value={0}
  />
  <Text
    style={
      Array [
        Object {
          "color": "#949494",
          "fontSize": 15,
          "fontWeight": "bold",
        },
        Object {
          "marginLeft": 30,
          "marginTop": 15,
        },
      ]
    }
  >
    MONTHLY INVESTMENT
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 21,
        },
        Object {
          "alignSelf": "center",
          "paddingTop": 10,
        },
      ]
    }
  >
    $
    0
  </Text>
  <RCTSlider
    disabled={false}
    enabled={true}
    maximumTrackTintColor="#919191"
    maximumValue={7000}
    minimumTrackTintColor="#00CFCC"
    minimumValue={0}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onSlidingComplete={null}
    onStartShouldSetResponder={[Function]}
    onValueChange={[Function]}
    step={0}
    style={
      Array [
        Object {
          "height": 40,
        },
        Object {
          "alignSelf": "stretch",
          "height": 40,
          "marginLeft": 25,
          "marginRight": 25,
        },
      ]
    }
    thumbTintColor="#00008b"
    value={0}
  />
  <Text
    style={
      Array [
        Object {
          "color": "#949494",
          "fontSize": 15,
          "fontWeight": "bold",
        },
        Object {
          "marginLeft": 30,
          "marginTop": 15,
        },
      ]
    }
  >
    YEARS WITH UP&UP
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 21,
        },
        Object {
          "alignSelf": "center",
          "paddingTop": 10,
        },
      ]
    }
  >
    0
     years
  </Text>
  <RCTSlider
    disabled={false}
    enabled={true}
    maximumTrackTintColor="#919191"
    maximumValue={5}
    minimumTrackTintColor="#00CFCC"
    minimumValue={1}
    onChange={null}
    onResponderTerminationRequest={[Function]}
    onSlidingComplete={[Function]}
    onStartShouldSetResponder={[Function]}
    onValueChange={null}
    step={0}
    style={
      Array [
        Object {
          "height": 40,
        },
        Object {
          "alignSelf": "stretch",
          "height": 40,
          "marginLeft": 25,
          "marginRight": 25,
        },
      ]
    }
    thumbTintColor="#00008b"
    value={0}
  />
  <Text
    style={
      Array [
        Object {
          "color": "#949494",
          "fontSize": 15,
          "fontWeight": "bold",
        },
        Object {
          "marginLeft": 30,
          "marginTop": 15,
        },
      ]
    }
  >
    OWNERSHIP VALUE
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#00CFCC",
          "fontSize": 21,
        },
        Object {
          "alignSelf": "center",
          "fontSize": 28,
          "marginTop": 15,
        },
      ]
    }
  >
    $
    0
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#949494",
          "fontSize": 12,
        },
        Object {
          "marginLeft": 30,
          "marginRight": 30,
          "textAlign": "center",
        },
      ]
    }
  >
    Use your $
    0
     ownership stake as a down payment on any home, transfer it to your next up&up rental, or cash out.
  </Text>
</View>
`;

exports[`<App /> renders calculator component with props correctly 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#949494",
          "fontSize": 15,
          "fontWeight": "bold",
        },
        Object {
          "marginLeft": 20,
          "marginTop": 15,
        },
      ]
    }
  >
    MONTHLY PAYMENT
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 20,
        "marginTop": 7,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 21,
        }
      }
    >
      $
       rent +
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#00CFCC",
            "fontSize": 21,
          },
          Object {
            "marginLeft": 5,
          },
        ]
      }
    >
      $
      0
       investment
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#d3d3d3",
        "height": 1,
        "marginTop": 3,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#d3d3d3",
        "height": 1,
        "marginTop": 2,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 30,
        "marginTop": 35,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#949494",
            "fontSize": 15,
            "fontWeight": "bold",
          },
          Object {
            "textDecorationLine": "line-through",
            "textDecorationStyle": "solid",
          },
        ]
      }
    >
      SECURITY DEPOSIT
    </Text>
    <Text
      style={
        Object {
          "color": "#949494",
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
       → INITIAL INVESTMENT
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 21,
        },
        Object {
          "alignSelf": "center",
          "paddingTop": 10,
        },
      ]
    }
  >
    $
    0
  </Text>
  <RCTSlider
    disabled={false}
    enabled={true}
    maximumTrackTintColor="#919191"
    maximumValue={12000}
    minimumTrackTintColor="#00CFCC"
    minimumValue={0}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onSlidingComplete={null}
    onStartShouldSetResponder={[Function]}
    onValueChange={[Function]}
    step={0}
    style={
      Array [
        Object {
          "height": 40,
        },
        Object {
          "alignSelf": "stretch",
          "height": 40,
          "marginLeft": 25,
          "marginRight": 25,
        },
      ]
    }
    thumbTintColor="#00008b"
    value={0}
  />
  <Text
    style={
      Array [
        Object {
          "color": "#949494",
          "fontSize": 15,
          "fontWeight": "bold",
        },
        Object {
          "marginLeft": 30,
          "marginTop": 15,
        },
      ]
    }
  >
    MONTHLY INVESTMENT
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 21,
        },
        Object {
          "alignSelf": "center",
          "paddingTop": 10,
        },
      ]
    }
  >
    $
    0
  </Text>
  <RCTSlider
    disabled={false}
    enabled={true}
    maximumTrackTintColor="#919191"
    maximumValue={7000}
    minimumTrackTintColor="#00CFCC"
    minimumValue={0}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onSlidingComplete={null}
    onStartShouldSetResponder={[Function]}
    onValueChange={[Function]}
    step={0}
    style={
      Array [
        Object {
          "height": 40,
        },
        Object {
          "alignSelf": "stretch",
          "height": 40,
          "marginLeft": 25,
          "marginRight": 25,
        },
      ]
    }
    thumbTintColor="#00008b"
    value={0}
  />
  <Text
    style={
      Array [
        Object {
          "color": "#949494",
          "fontSize": 15,
          "fontWeight": "bold",
        },
        Object {
          "marginLeft": 30,
          "marginTop": 15,
        },
      ]
    }
  >
    YEARS WITH UP&UP
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 21,
        },
        Object {
          "alignSelf": "center",
          "paddingTop": 10,
        },
      ]
    }
  >
    0
     years
  </Text>
  <RCTSlider
    disabled={false}
    enabled={true}
    maximumTrackTintColor="#919191"
    maximumValue={5}
    minimumTrackTintColor="#00CFCC"
    minimumValue={1}
    onChange={null}
    onResponderTerminationRequest={[Function]}
    onSlidingComplete={[Function]}
    onStartShouldSetResponder={[Function]}
    onValueChange={null}
    step={0}
    style={
      Array [
        Object {
          "height": 40,
        },
        Object {
          "alignSelf": "stretch",
          "height": 40,
          "marginLeft": 25,
          "marginRight": 25,
        },
      ]
    }
    thumbTintColor="#00008b"
    value={0}
  />
  <Text
    style={
      Array [
        Object {
          "color": "#949494",
          "fontSize": 15,
          "fontWeight": "bold",
        },
        Object {
          "marginLeft": 30,
          "marginTop": 15,
        },
      ]
    }
  >
    OWNERSHIP VALUE
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#00CFCC",
          "fontSize": 21,
        },
        Object {
          "alignSelf": "center",
          "fontSize": 28,
          "marginTop": 15,
        },
      ]
    }
  >
    $
    0
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#949494",
          "fontSize": 12,
        },
        Object {
          "marginLeft": 30,
          "marginRight": 30,
          "textAlign": "center",
        },
      ]
    }
  >
    Use your $
    0
     ownership stake as a down payment on any home, transfer it to your next up&up rental, or cash out.
  </Text>
</View>
`;

exports[`<App /> renders page correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "marginBottom": 20,
        "marginLeft": 10,
      }
    }
  >
    Component called with:: rent: 2000, Initial min investment: 1200, initial max: 12000, monthly max investment: 600, years max: 5
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#d3d3d3",
        "height": 7,
        "marginTop": 1,
      }
    }
  />
  <RCTScrollView>
    <View>
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#949494",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "marginLeft": 20,
                "marginTop": 15,
              },
            ]
          }
        >
          MONTHLY PAYMENT
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginLeft": 20,
              "marginTop": 7,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000000",
                "fontSize": 21,
              }
            }
          >
            $
            2000
             rent +
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#00CFCC",
                  "fontSize": 21,
                },
                Object {
                  "marginLeft": 5,
                },
              ]
            }
          >
            $
            0
             investment
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#d3d3d3",
              "height": 1,
              "marginTop": 3,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#d3d3d3",
              "height": 1,
              "marginTop": 2,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginLeft": 30,
              "marginTop": 35,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#949494",
                  "fontSize": 15,
                  "fontWeight": "bold",
                },
                Object {
                  "textDecorationLine": "line-through",
                  "textDecorationStyle": "solid",
                },
              ]
            }
          >
            SECURITY DEPOSIT
          </Text>
          <Text
            style={
              Object {
                "color": "#949494",
                "fontSize": 15,
                "fontWeight": "bold",
              }
            }
          >
             → INITIAL INVESTMENT
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 21,
              },
              Object {
                "alignSelf": "center",
                "paddingTop": 10,
              },
            ]
          }
        >
          $
          0
        </Text>
        <RCTSlider
          disabled={false}
          enabled={true}
          maximumTrackTintColor="#919191"
          maximumValue={12000}
          minimumTrackTintColor="#00CFCC"
          minimumValue={1200}
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onSlidingComplete={null}
          onStartShouldSetResponder={[Function]}
          onValueChange={[Function]}
          step={0}
          style={
            Array [
              Object {
                "height": 40,
              },
              Object {
                "alignSelf": "stretch",
                "height": 40,
                "marginLeft": 25,
                "marginRight": 25,
              },
            ]
          }
          thumbTintColor="#00008b"
          value={0}
        />
        <Text
          style={
            Array [
              Object {
                "color": "#949494",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "marginLeft": 30,
                "marginTop": 15,
              },
            ]
          }
        >
          MONTHLY INVESTMENT
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 21,
              },
              Object {
                "alignSelf": "center",
                "paddingTop": 10,
              },
            ]
          }
        >
          $
          0
        </Text>
        <RCTSlider
          disabled={false}
          enabled={true}
          maximumTrackTintColor="#919191"
          maximumValue={600}
          minimumTrackTintColor="#00CFCC"
          minimumValue={0}
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onSlidingComplete={null}
          onStartShouldSetResponder={[Function]}
          onValueChange={[Function]}
          step={0}
          style={
            Array [
              Object {
                "height": 40,
              },
              Object {
                "alignSelf": "stretch",
                "height": 40,
                "marginLeft": 25,
                "marginRight": 25,
              },
            ]
          }
          thumbTintColor="#00008b"
          value={0}
        />
        <Text
          style={
            Array [
              Object {
                "color": "#949494",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "marginLeft": 30,
                "marginTop": 15,
              },
            ]
          }
        >
          YEARS WITH UP&UP
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 21,
              },
              Object {
                "alignSelf": "center",
                "paddingTop": 10,
              },
            ]
          }
        >
          0
           years
        </Text>
        <RCTSlider
          disabled={false}
          enabled={true}
          maximumTrackTintColor="#919191"
          maximumValue={5}
          minimumTrackTintColor="#00CFCC"
          minimumValue={1}
          onChange={null}
          onResponderTerminationRequest={[Function]}
          onSlidingComplete={[Function]}
          onStartShouldSetResponder={[Function]}
          onValueChange={null}
          step={0}
          style={
            Array [
              Object {
                "height": 40,
              },
              Object {
                "alignSelf": "stretch",
                "height": 40,
                "marginLeft": 25,
                "marginRight": 25,
              },
            ]
          }
          thumbTintColor="#00008b"
          value={0}
        />
        <Text
          style={
            Array [
              Object {
                "color": "#949494",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "marginLeft": 30,
                "marginTop": 15,
              },
            ]
          }
        >
          OWNERSHIP VALUE
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#00CFCC",
                "fontSize": 21,
              },
              Object {
                "alignSelf": "center",
                "fontSize": 28,
                "marginTop": 15,
              },
            ]
          }
        >
          $
          0
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#949494",
                "fontSize": 12,
              },
              Object {
                "marginLeft": 30,
                "marginRight": 30,
                "textAlign": "center",
              },
            ]
          }
        >
          Use your $
          0
           ownership stake as a down payment on any home, transfer it to your next up&up rental, or cash out.
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
